walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventList = new ArrayList(events);
+
+    Collections.sort(eventList, new Comparator<Event>() {
+      @Override
+      public int compare(Event event1, Event event2) {
+        return event1.getWhen().start() - event2.getWhen().start();
+      }
+    });
+
+    int currentRangeStart = TimeRange.START_OF_DAY;
+    int currentRangeEnd;
+    int currentRangeDuration;
+
+    boolean sharesAttendees;
+
+    List<TimeRange> availableTimes = new ArrayList<>();
+
+    for (Event event : eventList) {
+      sharesAttendees = false;
+
+      for (String person : event.getAttendees()) {
+        if (request.getAttendees().contains(person)) {
+          sharesAttendees = true;
+          break;
+        }
+      }
+
+      if (!sharesAttendees) continue;
+
+      currentRangeEnd = event.getWhen().start();
+      currentRangeDuration = currentRangeEnd - currentRangeStart;
+
+      if (currentRangeEnd >= currentRangeStart) {
```

Comment:
Is it worth checking equality here? Will there ever be a duration request of '0'? If this is to satisfy that edge case, perhaps it's worth special-casing it so it's more apparent to someone reading the algorithm that this unexpected input is handled.

Reply:
It is not so much about a request duration of zero, but rather the case when event B starts right when event A ends, meaning that there is no space between them, but it indeed is more readable to have it as a separate case.